text/0119-dynamic-component-creation.md

```diff
@@ -0,0 +1,415 @@
+
+_This is similar to the [Angular approach to dynamic components](https://angular.io/guide/dynamic-component-loader)._
+
+#### Comparison with `<lwc:component lwc:is={expression}>`
+
+With `LightningDynamicElement` the framework keeps track of references to the dynamic components and also manages its lifecycle.  In contrast with `<lwc:component>` the component author is responsible for keeping track of the custom element.  This means that any dynamic component must be retrieved each time the component changes.  Additionally, any imperatively assigned attributes or event listeners will need to be reapplied once the component changes as well.
+
+## Adoption strategy
+
+This is a breaking change for existing user of `lwc:dynamic` as it supersedes the directive.  However, the `lwc:dynamic` directive has not reached GA and all consumers of the directive are internal.  They will need to switch their components to change the tag names to `lwc:component` and the directives from `lwc:dynamic` to `lwc:is`.
+
+The behavior of `lwc:is` is the same as `lwc:dynamic` when the constructor is not provided, both will not render the dynamic component of its children. The behavior is different when the components are rendered though, as `lwc:is` will render the custom element name rather than the custom element name used with `lwc:dynamic`.  Component authors may need to adjust any testing that relies on this.
+
+To help with the transition, the LWC team will support both `lwc:dynamic` and `lwc:is` directives for existing consumers.  Going forward, the compiler will report a warning when usage of the `lwc:dynamic` directive is detected until it is fully deprecated.
+
+# How we teach this
+
+This concept is fairly intuitive and is similar to the same concepts in other frameworks (Svelte, Vue, Angular).  Providing documentation around how the to use `<lwc:component>` and `lwc:is` with a small playground example should be sufficient.
+
+Examples of [documentation](https://svelte.dev/docs#template-syntax-svelte-component) and [playground](https://svelte.dev/tutorial/svelte-component) from Svelte.
+
+# Unresolved questions
+
+Will there be issues with collisions?  For example if a dynamic constructor is imported but has the same custom element name as a custom element that's already on the page?
+- Pivots in Locker and the `UpgradeableConstructor` should take care of this.
+
+Should we log errors when component authors try to use `<lwc:*>` incorrectly?  For example should we prevent component authors from doing something like `document.createElement('lwc:other');`?
+- In order to do this we would need to globally patch certain APIs which is something that we want to avoid to prevent bugs that are hard to maintain in the long run.  To handle this case, explicitly mention in the documentation that `<lwc:dynamic>` is an LWC compiler signal, and doesn't have any bearing at runtime.
```

Comment:
I'm fine with this... the idea really is to say that doing something like `document.createElement('lwc:other')` give you no powers.